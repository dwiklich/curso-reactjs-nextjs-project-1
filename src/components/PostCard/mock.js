export const postCardPropsMock = {
  cover: 'img/img.png',
  title: 'title 1',
  body: 'body 1',
  id: 1,
};

export const postsPropsMock = {
  posts: [
    {
      title: 'title post 1',
      body: 'body post 1',
      cover: 'img/img-post1.png',
      id: 1,
    },
    {
      title: 'title post 2',
      body: 'body post 2',
      cover: 'img/img-post2.png',
      id: 2,
    },
    {
      title: 'title post 3',
      body: 'body post 3',
      cover: 'img/img-post3.png',
      id: 3,
    },
    {
      title: 'title post 4',
      body: 'body post 4',
      cover: 'img/img-post4.png',
      id: 4,
    },
    {
      title: 'title post 5',
      body: 'body post 5',
      cover: 'img/img-post5.png',
      id: 5,
    },
  ],
};
