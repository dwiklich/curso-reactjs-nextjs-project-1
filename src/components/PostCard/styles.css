.post {
  background-color: #fff;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  transition: transform 200ms ease-in-out;
}

.post:hover {
  transform: scale(1.05);
}

.post-content {
  padding: 30px;
}

.photo-card {
  display: flex;
  justify-content: center;
  align-items: center;
}

.photo {
  max-width: 100%;
  max-height: 100%;
  background-size: cover;
}
