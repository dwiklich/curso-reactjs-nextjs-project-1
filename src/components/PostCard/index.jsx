import "./styles.css";

export const PostCard = ({ cover, title, body }) => {
  return (
    <div className="post">
      <div className="photo-card">
        <img src={cover} alt={title} className="photo" />
      </div>
      <div className="post-content">
        <h2>{title}</h2>
        <p>{body}</p>
      </div>
    </div>
  );
};
