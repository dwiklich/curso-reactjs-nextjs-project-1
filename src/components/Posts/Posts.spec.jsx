import { render, screen } from '@testing-library/react';
import { Posts } from '.';

import { postsPropsMock } from '../PostCard/mock';

const props = postsPropsMock;

describe('<Posts />', () => {
  test('should render Posts', () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole('heading', { name: /title post /i })).toHaveLength(5);
    expect(screen.getAllByRole('img', { name: /title post /i })).toHaveLength(5);
    expect(screen.getAllByText(/body post/i)).toHaveLength(5);
    expect(screen.getByRole('img', { name: /title post 2/i })).toHaveAttribute('src', 'img/img-post2.png');
  });

  test('should not render posts', () => {
    render(<Posts />);
    expect(screen.queryByRole('heading', { name: /title post /i })).not.toBeInTheDocument();
  });

  test('should length posts is empty', () => {
    render(<Posts />);
    expect(screen.queryAllByRole('heading', { name: /title post /i })).toHaveLength(0);
  });

  test('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should name of prop "posts" value default equal []', () => {
    const { container, debug } = render(<Posts />);

    expect(screen.queryAllByRole('heading')).toHaveLength(0);
    expect(screen.queryAllByRole('img')).toHaveLength(0);
    expect(screen.queryAllByText(/body post/i)).toHaveLength(0);
    expect(screen.queryByRole('heading')).not.toBeInTheDocument();
    expect(screen.queryByRole('img')).not.toBeInTheDocument();
    expect(container.querySelector('.posts')).toHaveClass('posts');
    expect(container.querySelector('.posts')).toBeInTheDocument();
    debug();
  });
});
