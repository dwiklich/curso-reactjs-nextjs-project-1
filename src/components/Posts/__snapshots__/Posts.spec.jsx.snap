// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <div
      class="photo-card"
    >
      <img
        alt="title post 1"
        class="photo"
        src="img/img-post1.png"
      />
    </div>
    <div
      class="post-content"
    >
      <h2>
        title post 1
         
        1
      </h2>
      <p>
        body post 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <div
      class="photo-card"
    >
      <img
        alt="title post 2"
        class="photo"
        src="img/img-post2.png"
      />
    </div>
    <div
      class="post-content"
    >
      <h2>
        title post 2
         
        2
      </h2>
      <p>
        body post 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <div
      class="photo-card"
    >
      <img
        alt="title post 3"
        class="photo"
        src="img/img-post3.png"
      />
    </div>
    <div
      class="post-content"
    >
      <h2>
        title post 3
         
        3
      </h2>
      <p>
        body post 3
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <div
      class="photo-card"
    >
      <img
        alt="title post 4"
        class="photo"
        src="img/img-post4.png"
      />
    </div>
    <div
      class="post-content"
    >
      <h2>
        title post 4
         
        4
      </h2>
      <p>
        body post 4
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <div
      class="photo-card"
    >
      <img
        alt="title post 5"
        class="photo"
        src="img/img-post5.png"
      />
    </div>
    <div
      class="post-content"
    >
      <h2>
        title post 5
         
        5
      </h2>
      <p>
        body post 5
      </p>
    </div>
  </div>
</div>
`;
